src/util.h

```diff
@@ -46,6 +46,15 @@
                 UNIQ_T(A,aq) < UNIQ_T(B,bq) ? UNIQ_T(A,aq) : UNIQ_T(B,bq); \
         })
 
+
+/* "linux/fs.h" contains wrong definitions of FS_IOC_[GS]ETFLAGS.
+ * This problem has been known for at least 14 years. To avoid a spurious
+ * warning from valgrind, let's override the kernel definitions. */
+#undef FS_IOC_GETFLAGS
+#undef FS_IOC_SETFLAGS
+#define FS_IOC_GETFLAGS                 _IOR('f', 1, int)
+#define FS_IOC_SETFLAGS                 _IOW('f', 2, int)
```

Comment:
hmm, this appears wrong to me. the ioctl numbers are generated from a sizeof() of the last argument of _IOR()/_IOW(), and hence by making the above change you change the ioctl numbers, and that means on archs where sizeof(long) != sizeof(int) you will call non-existing ioctls...

valgrind appears broken if it doesn's special case this ioctl... There's no other way than special casing it, and it's what FUSE for example does...